backend/app/usecase/url/urlcreator_fake.go

```diff
@@ -14,13 +14,13 @@ type FakeCreator struct {
 }
 
 // CreateURL persists a new url with a generated alias in the repository.
-func (f FakeCreator) CreateURL(url entity.URL, userEmail string) (entity.URL, error) {
+func (f FakeCreator) CreateURL(url entity.URL, userEmail string, isPublic bool) (entity.URL, error) {
```

Comment:
@byliuyang I've considered moving `userEmail` and `isPublic` parameters into a separate struct, and an instance of that can instead be passed in as a parameter, but not sure if that'd make things any better. Your thoughts?

Reply:
The full codeclimate report also includes the call to `keyGen.NewKey()`, so perhaps this part can be refactored to a separate package-private function that generates random alias?

Reply:
@Coteh What would be the name of the struct?
@Coteh I'll exempt you from this refactor.

Reply:
@byliuyang I was thinking something along the lines of this (now accounting for changes that have been made since this PR was first posted):
```
type URLProperties struct {
    IsPublic bool
}
```
This can be a struct that comprises of properties that are not innate attributes of a URL itself, but are attributes of how the URL is to be present on short.

Another option would be to introduce the `IsPublic` property into the `URL` entity struct instead and populate it in the `URL` instance within the `CreateURL` mutation.

Reply:
@Coteh Let's open an issue to discuss everything related to this struct. I am leaning towards option 1 although the name of struct can be discuss. We can take care of this change in a separate PR.